src/Inciter/AMR/types.h

```diff
@@ -0,0 +1,48 @@
+#ifndef AMR_types_h
+#define AMR_types_h
+
+#include <array>
+#include <vector>
+#include <map>
+
+#include "../Base/Types.h"
+
+const int DIMENSION = 3;
+const size_t NUM_TET_NODES = 4;
+const size_t NUM_FACE_NODES = 3;
+const size_t NUM_TET_EDGES = 6;
+const size_t NUM_TET_FACES = 4;
+const size_t ID_SHIFT = 3;
+const size_t MAX_CHILDREN = 8;
+const char KEY_DELIM = '-';
+
+using real_t = tk::real;
+using coordinate_t = std::array<real_t, DIMENSION>;
+using tet_t = std::array<size_t, NUM_TET_NODES>;
+
+using node_pair_t  = std::array<std::size_t, 2>; // TODO: should this be a std::pair
```

Comment:
Not too bad as `std::array` either, because then the elements can be accessed as `[0]` and `[1]` instead of the more verbose `.first` and `.second`.